src/main/java/de/hs_mannheim/informatik/ct/controller/ContactTracingController.java

```diff
@@ -140,4 +237,26 @@ public String doSearch(@RequestParam String email, Model model) {
         private String header;
         private Function<Contact<?>, String> cellValue;
     }
+
+
+    /**
+     * Helper method to filter the contacts into 3 subcategories: students, employees of the university and guests
+     *
+     * @param contacts List of all possible contacts of the target email
+     * @param type     String of the filter type (students/ staff/ guests)
+     * @return filtered List of Contacts
+     */
+    private List<Contact<?>> filterContactList(Collection<Contact<?>> contacts, String type) {
+        List<Contact<?>> filteredList = new ArrayList<>();
+        for (Contact contact : contacts) {
+            if (type.equals("students") && contact.getContact().getEmail().contains("@stud.hs-mannheim.de")) {
+                filteredList.add(contact);
+            } else if (type.equals("staff") && contact.getContact().getEmail().contains("@hs-mannheim.de")) {
```

Comment:
Please make sure to add addresses with @lba.hs-mannheim (Lehrbeauftragte) here as well. Cf. PR #245